.github/scripts/ci-build.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+
+echo "## prepare environment ##"
+if [[ "${GITHUB_REF}" == "refs/heads/nightly" ]]; then
+  echo "-- NIGHTLY --"
+  export PUBLISH_BINARY="true"
+  export PUBLISH_DIR="nightlies"
+  export BUILD_TYPE="ReleaseWithDebInfo"
+  export VERSION_TAG="$(date +%Y%m%d)"
+elif [[ "${GITHUB_REF}" == "refs/tags/"* ]]; then
+  echo "-- RELEASE --"
+  export PUBLISH_BINARY="true"
+  export PUBLISH_DIR="releases"
+  export BUILD_TYPE="ReleaseWithDebInfo"
+elif [[ "${GITHUB_REF}" == "refs/pull/"* ]]; then
+  export PULL_REQUEST=$(echo "${GITHUB_REF}" | cut -d '/' -f 3)
+  echo "-- PULL REQUEST ${PULL_REQUEST} --"
+  export PUBLISH_BINARY="true"
+  export PUBLISH_DIR="pull-requests/${PULL_REQUEST}"
+  export VERSION_TAG="${PULL_REQUEST}pullrequest"
+  export BUILD_TYPE="Debug"
+else
+  echo "-- QUICK BUILD --"
+  export PUBLISH_BINARY="false"
+  export BUILD_TYPE="Debug"
+fi
+export BUILD_CMD="cmake --build . --config $BUILD_TYPE"
+export CONFIGURE_CMD="cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DVERSION_TAG=${VERSION_TAG}"
+if [[ "${BUILD_TYPE}" != "Debug" ]]; then
+   export CONFIGURE_CMD="${CONFIGURE_CMD} -DWITH_EDITOR_SLF=ON"
+fi
+if [[ "${SFTP_PASSWORD}" == "" ]]; then
+  export PUBLISH_BINARY="false"
+fi
+if [[ "$CI_TARGET" == "linux" ]]; then
+  sudo apt update
+  sudo apt install build-essential libsdl2-dev libfltk1.3-dev
+  # TODO tests fail with normal boost (libboost-filesystem-dev libboost-system-dev)
+  export CONFIGURE_CMD="${CONFIGURE_CMD} -DCMAKE_INSTALL_PREFIX=/usr -DEXTRA_DATA_DIR=/usr/share/ja2 -DLOCAL_BOOST_LIB=ON -DCPACK_GENERATOR=DEB"
+elif [[ "$CI_TARGET" == "mingw" ]]; then
+  sudo apt update
+  sudo apt install build-essential mingw-w64
+  export CONFIGURE_CMD="${CONFIGURE_CMD} -DCMAKE_TOOLCHAIN_FILE=./cmake/toolchain-mingw.cmake -DCPACK_GENERATOR=ZIP"
+elif [[ "$CI_TARGET" == "mac" ]]; then
+  export CONFIGURE_CMD="${CONFIGURE_CMD} -DCMAKE_TOOLCHAIN_FILE=./cmake/toolchain-macos.cmake -DCPACK_GENERATOR=Bundle"
+else
+  echo "unexpected target ${CI_TARGET}"
+  exit 1
+fi
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain=$(cat ./rust-toolchain) -y --profile minimal
+export PATH=$PATH:$HOME/.cargo/bin
+if [[ "$CI_TARGET" == "mingw" ]]; then
+  rustup target add x86_64-pc-windows-gnu
+else
+  rustup component add rustfmt
+  rustup component add clippy
+  cargo clippy -- -V
+  cargo fmt -- -V
+fi
+echo "PUBLISH_BINARY=${PUBLISH_BINARY}"
+echo "PUBLISH_DIR=${PUBLISH_DIR}"
+echo "BUILD_CMD=${BUILD_CMD}"
+echo "CONFIGURE_CMD=${CONFIGURE_CMD}"
+rustc -V
+cargo -V
+cmake --version
+
+echo "## configure, build, package ##"
+mkdir ci-build
+cd ci-build
+$CONFIGURE_CMD ..
+cat ./CMakeCache.txt
+$BUILD_CMD
+$BUILD_CMD --target package
+
+echo "## test ##"
+if [[ "$CI_TARGET" != "mingw" ]]; then
+  sudo $BUILD_CMD --target install
```

Comment:
Isn't that only required for "linux" builds?

Reply:
All of them should be tested if possible, mingw can't be tested because it's linux cross compiling to windows. I could be tested if it was built in a windows host.